File: baml_language/sdk_tests/crates/cpp/function_calls/customizable/tests/test_main.cpp
// Smoke tests for plain (non-LLM) expression functions.
// Port of python_pydantic2/function_calls/customizable/test_main.py.
#include <baml_sdk.hpp>
#include <baml_test.hpp>

BAML_TEST(hello_world_returns_literal) {
    BAML_ASSERT_EQ(baml_sdk::hello_world(), std::string("hello world"));
}

BAML_TEST(single_required_arg_round_trips) {
    // The next step up from the nullary case: one required positional
    // argument round-trips through the engine unchanged.
    BAML_ASSERT_EQ(baml_sdk::single_required_arg("hi"), std::string("hi"));
}

BAML_TEST_MAIN()

File: baml_language/sdk_tests/crates/swift/function_calls/customizable/TestMain.swift
// Smoke tests for plain (non-LLM) expression functions — port of
// python_pydantic2 `test_main.py`.
import XCTest
import Baml

final class TestMain: XCTestCase {
    func test_main_hello_world_returns_literal() throws {
        XCTAssertEqual(try Baml.hello_world(), "hello world")
    }

    func test_main_single_required_arg_round_trips() throws {
        // The next step up from the nullary case: one required argument
        // round-trips through the engine unchanged.
        XCTAssertEqual(try Baml.single_required_arg(value: "hi"), "hi")
    }
}
